src/lib/libdylink.js

```diff
@@ -959,11 +1014,24 @@ var LibraryDylink = {
      * @param {number=} handle
      * @param {Object=} localScope
      */`,
-  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, handle) {
+  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, rpath = {parentLibPath: '', paths: []}, handle) {
 #if DYLINK_DEBUG
     dbg(`loadDynamicLibrary: ${libName} handle: ${handle}`);
     dbg(`existing: ${Object.keys(LDSO.loadedLibsByName)}`);
 #endif
+
+#if FILESYSTEM
+      var runtimePathsAbs = (rpath.paths || []).map((p) => replaceORIGIN(rpath.parentLibPath, p));
+      var searchDirs = getDefaultLibDirs().concat(runtimePathsAbs);
+      var [existsInFS, libNameAbs] = locateLibraryFromFS(libName, searchDirs);
```

Comment:
If `locateLibraryFromFS` is changed to return just a single value, then the following replaces `libName` with the value returned from `locateLibraryFromFS` unless it was `null`:
```suggestion
      libName = locateLibraryFromFS(libName, searchDirs) ?? libName;
```

Reply:
But I guess you need `existsInFS` later, so the result from `locateLibraryFromFS` needs to be stored to calculate that.

Reply:
I think `readLibraryFromFS` and `locateLibraryFromFS` should be combined into one function. Then we can just do:
```
var libData = readLibraryFromFS(libName, searchDirs);
if (libData) {
   return flags.loadAsync ? Promise.resolve(libData) : libData;
}
```

Reply:
I wanted to call `libname = libnameAbs` at the very first part of the function because other wise, different `libname` will be used in `loadedLibsByName`, `mergeLibSymbols`, `sharedModules`.